file_writer.go

```diff
@@ -231,7 +232,10 @@ func (f *FileWriter) Close() error {
 
 	err := f.client.namenode.Execute("complete", completeReq, completeResp)
 	if err != nil {
-		return &os.PathError{"create", f.name, err}
+		return &os.PathError{Op: "close", Path: f.name, Err: err}
+	}
+	if *completeResp.Result != true {
+		return &os.PathError{Op: "close", Path: f.name, Err: errors.New("incomplete write")}
```

Comment:
I am not sure whether this is the preferred way to introduce a new error, though...